galaxy/SavePanel.gd

```diff
@@ -0,0 +1,69 @@
+extends Panel
+
+signal CloseSaveMenu
+var saves = []
+var current_selection = -1
+var button_ = ""
+
+
+func _ready():
+	refresh_saves()
+
+
+func _input(event):
+	if event.is_action_pressed("pause"):
+		_on_CancelButton_pressed()
+
+
+func refresh_saves():
+	while $ItemList.get_item_count() > 0:
+		$ItemList.remove_item(0)
+	for save_name in saves:
+		$ItemList.add_item(save_name)
+
+
+func add_new_save(save_name):
+	saves.append(save_name)
+	refresh_saves()
+
+func delete_save(index):
+	$ItemList.remove_item(index)
+
+func load_save():
+	pass
+	
+func overwrite_save():
+	pass
+
+func _on_CancelButton_pressed():
+	get_node("SaveNameInput").text = ""
+	emit_signal("CloseSaveMenu")
+
+
+func _on_NewSaveButton_pressed():
+	var new_save_name = get_node("SaveNameInput").text
+	if new_save_name != "":
+		add_new_save(new_save_name)
+
+
+func _on_OverwriteSaveButton_pressed():
+	button_ = "S"
+	if current_selection > -1:
+		get_parent().confirmation_popup("Are you sure you want to overwrite this save?", self)
+			
+	
+func _on_DeleteButton_pressed():
+	button_ = "D"
+	if current_selection > -1:
+		get_parent().confirmation_popup("Are you sure you want to delete this save?", self)
```

Comment:
What's with the `button_ = "D"`?
If that's for the user to press. I'd say it's better to add it as an input map under `Project > Project settings... > Input Map`, that way it can be easily changed when we need to. Plus anyone will be able to set up their own keymaps.

Reply:
That's to do with remembering which popup is open. Because popups are handled by the main UI code, i need to store which one i requested so when it comes back i know what action to take. D stand for Delete and the other option is S for save overwriting.

Reply:
Makes sense.
In that case:
1. Seems to me like it would make more sense to use an enum.
2. Maybe have a more self explanatory name for the variable, like `current_action` or whatever else you see fit.